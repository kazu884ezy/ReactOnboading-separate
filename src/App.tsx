import React from 'react';
import Alert from '@material-ui/lab/Alert';
import { useSelector } from 'react-redux';
import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
import OverTime from "./pages/OverTime";
import SpecificDate from "./pages/SpecificDate";
import GlobalMenu from "./layouts/GlobalMenu";
import NotFound from "./pages/NotFound";
import Spinner from "./components/Spinner";
import {makeStyles} from "@material-ui/core/styles";

const useStyles = makeStyles({
   alert: {
       maxWidth: '600px',
       margin: '20px auto',
       borderRadius: '5px'
   }
});

function App() {
    const loading = useSelector(state => state.common.loading);
    const error = useSelector(state => state.common.error);

    const classes = useStyles();

  return (
      <div className="App">
        <Router>
          <GlobalMenu />
          <div className="main">
              { loading && <Spinner fixed={true} />}
              { error && <Alert className={classes.alert} severity="error">{error}</Alert>}
            <Switch>
              <Route path="/exchange-rates-at-a-specific-date" component={SpecificDate} />
                <Route path="/exchange-rates-overtime" component={OverTime} />
                <Route path="/" exact component={SpecificDate} />
              <Route component={NotFound} />
            </Switch>
          </div>
        </Router>
      </div>
  );
}

export default App;
