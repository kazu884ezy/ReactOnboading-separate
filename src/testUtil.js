import store from './store';

export const makeTestStore = () => {
    const origDispatch = store.dispatch;
    store.dispatch = jest.fn(origDispatch);
    return store;
};