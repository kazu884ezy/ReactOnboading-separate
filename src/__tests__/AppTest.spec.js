import React from 'react';
import {render, screen} from '@testing-library/react';
import App from "../App";
import {makeTestStore} from "../testUtil";
import {Provider} from "react-redux";

describe('App page test', () => {
    test('render correctly', () => {
        const store = makeTestStore();

        const { container } = render(<Provider store={store}><App /></Provider>);
        expect(container.firstChild).toMatchSnapshot();
    })
})