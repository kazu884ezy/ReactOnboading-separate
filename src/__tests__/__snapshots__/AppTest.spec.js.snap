// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page test render correctly 1`] = `
<div
  class="App"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Specific Date
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Over Time
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-2 PrivateTabIndicator-colorSecondary-4 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </header>
  <div
    class="main"
  >
    <div
      class="makeStyles-root-9 makeStyles-fix-11"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div>
      <h1
        class="makeStyles-heading-6"
      >
        Exchange Rates for a Day
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                role="button"
                tabindex="0"
              >
                NZD
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="NZD"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                type="text"
                value="11/15/2020"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="makeStyles-main-7"
      >
        <table
          class="MuiTable-root makeStyles-table-8"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Currency
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>
</div>
`;
